### DISCUSSION TITLE
Version Upgrade from 5.0.3 to 5.0.5

### DISCUSSION THREAD

**Discussion by gprao08** [Sunbird-cQube/community#843] [General]
@bommalla @RNarmatha
Could you plz let me know before version upgrade the below commands is enough to run or more?
step: 1
sudo docker stop $(sudo docker ps -aq)
sudo docker rm $(sudo docker ps -aq)
sudo docker rmi $(sudo docker images -q)
sudo docker volume prune -y
sudo docker volume prune
sudo docker volume rm adapter_volume monitoring_grafana_data monitoring_prometheus_data python_executable_env
sudo docker network prune

sudo rm -rf /opt/cqube
sudo rm -rf /opt/containerd

sudo rm -rf ~/cqube-devops
step:2
Clone repo again by using below command
git clone https://github.com/Sunbird-cQube/cqube-devops.git
git checkout release v5.0.3
git pull
git branch
Then you can start the installation
sudo ./install.sh

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Yes sir same steps you can follow, while checkout the release branch instead of v5.0.3 to put v5.0.5.

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
While installing 5.0.5 it is asking SSL file name and key
cQube/community/assets/143190553/7fe2cda2-8517-4847-a7bf-f6f48602df28)
 file name and the path is showing: ansible/ssl_certificates. I skip the two names and continue installation. Can I do the job after installation?
Screen shot attached. 
![additional requirements for version 5 0 5](https://github.com/Sunbird-
![pgramme_selector file](https://github.com/Sunbird-cQube/community/assets/143190553/78784a4e-c4a7-4d32-9f30-b27678f640a4)

**Reply by gprao08** [reply]
![additional requirements for version 5 0 5](https://github.com/Sunbird-cQube/community/assets/143190553/4d2c865e-a2c5-4abe-b68b-43250bb5338b)

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Before going to start the installation we need move the SSL keys to below mentioned directory

/root/cqube-devops/ansible/ssl_certificates  

Please fin the below document for your reference.
https://cqube.sunbird.org/deployment-process/sdc-deployment

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
thanks for you reply. After install 5.0.5 and as you said I copy the ssl files into the path and configure keyclock admin console, it is redirecting to miniobucket instead of dashboard. Can you help me out plz.
I gave URL https://vsk.ap.gov.in/   it is redirecting to https://vsk.ap.gov.in/minio/ui/login and asking login into miniobucket

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Please give the below URL in the browser

https://vsk.ap.gov.in/dashboard/

**Reply by gprao08** [reply]
@bommalla 
Is it not possible if hit only vsk.ap.gov.in to redirect to https://vsk.ap.gov.in/dashboard/login
If I hit https://vsk.ap.gov.in/dashboard           it is directly opening dashboard without asking login credintials

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
In v5.0.5 there is two types users is there one is public access and other one is private access.
If you have public access please hit  this url in the browser https://vsk.ap.gov.in/dashboard
if you want private access please hit this url in the browser https://vsk.ap.gov.in/dashboard/login

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Any update on the ingestion ?

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
Department provided new VSK data and they asked me to ingest the latest data only. Now I am going to check the data and will ask you if any doubts. Requesting your support in this regard. Tq

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Thanks for the update!!

**Comment by gprao08** [comment]
@bommalla @RNarmatha 
After install 5.0.5 I tried to run dimension_scheema run collection from postman. But it show error (error response code 400). Screen shot attached for your reference. Could you plz help me to resolve the issue.
![image](https://github.com/Sunbird-cQube/community/assets/143190553/324b830b-9478-4a9b-9376-482ef78c1245)

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

I think already schema is there in the database please check the schema in the database. if schema is there please nuke the database and re run the dimension schema 

Command : yarn cli nuke-db

**Reply by gprao08** [reply]
@bommalla 
Now I update version 5.0.5, I could not found the actual path. Could you plz mention the path where to run the **yarn cli nuke-db** command?

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

This is fresh installation right ?

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
Yes

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

sudo docker exec -it nifi_app bash

cd /Sunbird-cQube-processing-ms/impl/c-qube/ingest

Then you can run the below command 

yarn cli nuke-db

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
Thanks, This issue resolved now and shows 15 event rows and 5 dimension rows identified in DB

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
Thanks for the update

**Comment by gprao08** [comment]
@bommalla @RNarmatha 
Hi Rajesh, After I select the dashboard it redirected to logout after I successfully run adapter and run latest code local. Moreover I found the below files (attached screen shot) in dimensions folder. Could you plz support to resolve the issue...
![image](https://github.com/Sunbird-cQube/community/assets/143190553/440f08dd-3c93-48e1-9fca-6ef8aaecf038)

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Release v5.0.5 ingestion is little different please find the below deployment document and do the ingestion as per document 

https://cqube.sunbird.org/deployment-process/sdc-deployment

**Comment by gprao08** [comment]
@bommalla @RNarmatha 
Hi Rajesh
After reading the above document I have missed some steps those are Step 5 (VSK dimensions to the said path), and in data ingestion I used 'run adapter' and 'run code local' instead of 'data_moving_local' and 'ingest_data'.
What shall I do, shall I start from beginning? or can we resolve?

**Comment by RNarmatha** [comment]
Hi,

You can continue from the processing
*Regards*





*Narmatha R*

Project Manager


*M* +91 9600095667

*E* ***@***.*** ***@***.***>*

[image: LinkedIn] <https://www.linkedin.com/company/1913934> [image:
Facebook] <https://www.facebook.com/TibilSolutions> [image: Twitter]
<https://twitter.com/TibilSolutions> [image: Insta]
<https://www.instagram.com/tibilsolutions/>
*This message (including any attachments) may contain confidential,
proprietary, privileged, and/or private information. The information is
intended to be for the use of the individual or entity designated above. If
you are not the intended recipient of this message, please notify the
sender immediately, and delete the message and any attachments. Any
disclosure, reproduction, distribution, or other use of this message or any
attachments by an individual or entity other than the intended recipient is
prohibited.**t is prohibited.*

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
After click any of the dashboard it automatically logged out & Only it allows the URL https://vsk.ap.gov.in/dashboard/#/login
If I give https://vsk.ap.gov.in it redirect to https://vsk.ap.gov.in/minio/ui/login

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

All the files is there in minio bucket then you can schedule the 'run adapter'. after that you can scheduled  data_moving_local'

**Reply by gprao08** [reply]
Yes all the files uploaded without errors.
![image](https://github.com/Sunbird-cQube/community/assets/143190553/eca61f80-9c98-4ba9-8131-152d6bc3a8dc)

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
After that you can schedule the one by one program by using schedule API.

Body:
              {
                 "processor_group_name": "ingest_data",
                  "scheduled_at": "0 31 14 * * ?",
                  "program_name": "nishtha"
               }

Note: after nishatha you can schedule pgi,nas,pmposhan ....etc

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
Just remove the login from the url. please hit the below url in the browser with login

https://vsk.ap.gov.in/dashboard

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Login issue fixed please check now

**Reply by gprao08** [reply]
Yes. Thanks

**Comment by gprao08** [comment]
@bommalla @RNarmatha 
Hi Rajesh Even after I run all programs individually only UDISE+ dashboard displayed remaining shows 0 values

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Again we can schedule rest other programs,each program we can give 2min time diff and once the program visualise in dashboard then only can run other program
